table/tables/index.go

```diff
@@ -159,12 +173,7 @@ func (c *index) GenIndexKey(sc *stmtctx.StatementContext, indexedValues []types.
 			}
 		}
 	}
-
-	if c.buffer != nil {
-		key = c.buffer[:0]
-	} else {
-		key = make([]byte, 0, len(c.prefix)+len(indexedValues)*9+9)
-	}
+	key = c.getIndexKeyBuf(buf, len(c.prefix)+len(indexedValues)*9+9)
```

Comment:
This function is not only used in DDL. So is it safe to use the c.buffer in this function?

Reply:
This function is not only used in DDL, but c.buffer is only set in DDL. so c.buffer will not be used by others.